api/proto/teleport/inventory/v1/inventory_service.proto

```diff
@@ -0,0 +1,100 @@
+
+// InventoryService provides methods to list instances from the inventory cache. This service does not fallback to the backend if the cache is unhealthy.
+service InventoryService {
+  // ListUnifiedInstances returns a page of teleport instances and bot_instances. This API will refuse any requests when the cache is unhealthy or not yet
+  // fully initialized.
+  rpc ListUnifiedInstances(ListUnifiedInstancesRequest) returns (ListUnifiedInstancesResponse);
+}
+
+// UnifiedInstanceSort specifies the sort mode for listing unified instances.
+enum UnifiedInstanceSort {
+  UNIFIED_INSTANCE_SORT_UNSPECIFIED = 0;
+  // UNIFIED_INSTANCE_SORT_NAME sorts by display name (hostname for instances, bot name for bot instances).
+  UNIFIED_INSTANCE_SORT_NAME = 1;
+  // UNIFIED_INSTANCE_SORT_TYPE sorts by instance type (instance vs bot_instance).
+  UNIFIED_INSTANCE_SORT_TYPE = 2;
+  // UNIFIED_INSTANCE_SORT_VERSION sorts by version.
+  UNIFIED_INSTANCE_SORT_VERSION = 3;
+}
+
+// SortOrder specifies the sort order for listing unified instances.
+enum SortOrder {
+  SORT_ORDER_UNSPECIFIED = 0;
+  // SORT_ORDER_ASCENDING sorts in ascending order.
+  SORT_ORDER_ASCENDING = 1;
+  // SORT_ORDER_DESCENDING sorts in descending order.
+  SORT_ORDER_DESCENDING = 2;
+}
+
+// ListUnifiedInstancesRequest is the request for listing instances.
+message ListUnifiedInstancesRequest {
+  // page_size is the size of the page to return.
+  int32 page_size = 1;
+  // page_token is the next_page_token value returned from a previous ListUnifiedInstances request, if any.
+  string page_token = 2;
+  // filter specifies optional search criteria to limit which instances should be returned.
+  ListUnifiedInstancesFilter filter = 3;
+  // sort specifies the sort mode for the results. Defaults to UNIFIED_INSTANCE_SORT_NAME.
+  UnifiedInstanceSort sort = 4;
+  // order specifies the sort order for the results. Defaults to SORT_ORDER_ASCENDING.
+  SortOrder order = 5;
+}
+
+// ListUnifiedInstancesResponse is the response from listing instances.
+message ListUnifiedInstancesResponse {
+  // items is the list of instances (instances or bot instances) returned.
+  repeated UnifiedInstanceItem items = 1;
+  // next_page_token contains the next page token to use as the start key for the next page of instances.
+  string next_page_token = 2;
+}
+
+// ListUnifiedInstancesFilter provides a mechanism to refine ListUnifiedInstances results.
+message ListUnifiedInstancesFilter {
+  // search is a basic string search query which will filter results by name (hostname for instances, bot name for bot instances).
+  string search = 1;
+  // search_keywords is a list of search keywords using predicate language to match against resource field values.
+  string search_keywords = 2;
+  // instance_types is the types of instances to return. If omitted, both instances and bot instances will be returned.
+  repeated string instance_types = 3;
```

Comment:
List the allowed values? We could also make this an enum rather than use a string I suppose.